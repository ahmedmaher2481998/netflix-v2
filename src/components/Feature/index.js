import { Container, Break, Title, SubTitle } from './styles/Feature';
import React from 'react';

const Feature = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Feature.Title = function FeatureTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Feature.Break = function FeatureBreak({ children, ...restProps }) {
  return <Break {...restProps}>{children}</Break>;
};

Feature.SubTitle = function FeatureSubTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

export default Feature;
