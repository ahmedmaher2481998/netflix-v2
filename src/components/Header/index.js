import { Background, ButtonLink, Frame, Logo } from './styles/Header';

const Header = ({ bg, Children, ...restProps }) => {
  return bg ? <Background {...restProps}>{Children}</Background> : Children;
};

Header.Logo = function HeaderLogo({ children, ...restProps }) {
  return <Logo {...restProps}>{children}</Logo>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>;
};
Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

export default Header;
