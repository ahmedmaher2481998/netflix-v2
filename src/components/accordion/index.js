import { createContext, useContext, useState } from 'react';
import { Container, Inner, Body, Frame, Header, Item, Title } from './styles/accordion';

const ShowToggleContext = createContext();

function Accordion({ children, ...resProps }) {
  return (
    <>
      <Container>
        <Inner {...resProps}>{children}</Inner>
      </Container>
    </>
  );
}

Accordion.Item = function AccordionItem({ children, ...resProps }) {
  const [showToggle, setShowToggle] = useState(false);

  return (
    <ShowToggleContext.Provider value={{ showToggle, setShowToggle }}>
      <Item {...resProps}> {children}</Item>
    </ShowToggleContext.Provider>
  );
};

Accordion.Frame = function AccordionFrame({ children, ...resProps }) {
  return <Frame {...resProps}> {children}</Frame>;
};

Accordion.Header = function AccordionHeader({ children, ...resProps }) {
  const { setShowToggle, showToggle } = useContext(ShowToggleContext);
  return (
    <Header onClick={() => setShowToggle((pre) => !pre)} {...resProps}>
      {children}
      {showToggle ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = function AccordionBody({ children, ...resProps }) {
  const { showToggle } = useContext(ShowToggleContext);
  return showToggle ? <Body {...resProps}> {children}</Body> : null;
};

Accordion.Title = function AccordionTitle({ children, ...resProps }) {
  return <Title {...resProps}> {children}</Title>;
};

export default Accordion;
