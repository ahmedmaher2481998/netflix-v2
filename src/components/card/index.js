import React from 'react';
import {
  Container,
  Title,
  SubTitle,
  Item,
  Image,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Group,
  Text,
  Entities,
  Meta,
  Maturity,
  Content
} from './styles/cards';
import { useContext, createContext, useState } from 'react';
const FeatureContext = createContext();
const Card = ({ children, ...rest }) => {
  const [showFeature, setShowFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState({});
  return (
    <FeatureContext.Provider
      value={{
        showFeature,
        setShowFeature,
        itemFeature,
        setItemFeature
      }}></FeatureContext.Provider>
  );
};

export default Card;
