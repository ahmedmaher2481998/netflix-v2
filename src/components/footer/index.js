import React from "react";
import {
	Title,
	Row,
	Column,
	Container,
	Link,
	Text,
	Break,
} from "./styles/footer";
const Footer = () => {
	return <div>Footer</div>;
};
Footer.Container = ({ children, ...resProps }) => (
	<Container {...resProps}>{children}</Container>
);

Footer.Title = ({ children, ...resProps }) => (
	<Title {...resProps}> {children}</Title>
);
Footer.Row = ({ children, ...resProps }) => (
	<Row {...resProps}> {children}</Row>
);

Footer.Column = ({ children, ...resProps }) => (
	<Column {...resProps}> {children}</Column>
);

Footer.Link = ({ children, ...resProps }) => (
	<Link {...resProps}> {children}</Link>
);

Footer.Text = ({ children, ...resProps }) => (
	<Text {...resProps}> {children}</Text>
);

Footer.Break = ({ children, ...resProps }) => (
	<Break {...resProps}> {children}</Break>
);

export default Footer;
