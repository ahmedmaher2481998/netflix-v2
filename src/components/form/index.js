import React from 'react';
import { Link, Error, Container, Text, SmallText, Input, Submit, Title, Base } from './styles/form';
const Form = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

export default Form;
