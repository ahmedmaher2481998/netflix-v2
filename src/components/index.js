export { default as JumboTron } from './jumbotron/index';
export { default as Footer } from './footer/index';
export { default as Accordion } from './accordion/index';
