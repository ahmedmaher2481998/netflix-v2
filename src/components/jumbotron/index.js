import React from 'react';
import { Inner, Item, Pane, Container, Image, Subtitle, Title } from './styles/jumbotron';

function JumboTron({ children, direction = 'row', ...restProps }) {
  return (
    <Item {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
}

JumboTron.Container = function JumboTronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

JumboTron.Pane = function JumboTronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

JumboTron.Title = function JumboTronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

JumboTron.Subtitle = function JumboTronSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

JumboTron.Image = function JumboTronImage({ ...restProps }) {
  return <Image {...restProps} />;
};

export default JumboTron;
