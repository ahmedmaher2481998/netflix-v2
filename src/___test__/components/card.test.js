import { render, fireEvent } from '@testing-library/react';
import { Card, Player } from '../../components';
const category = 'series';
const slideRow = [
  {
    title: 'Documentaries',
    data: [
      {
        genre: 'documentaries',
        maturity: '18',
        slug: 'tiger-king',
        description: 'Tiger King description',
        id: '07adcf19-517f-41b4-8f8b-04fee694bce1',
        title: 'Tiger King',
        docId: 'IcRxJBanzE7nmIp2GMGb'
      }
    ]
  },
  {
    title: 'Feel Good',
    data: [
      {
        title: 'Juno',
        genre: 'feel-good',
        description: 'Juno description',
        maturity: '0',
        id: 'a938b5a6-fe25-4a06-8050-353bc7146ccd',
        slug: 'juno',
        docId: '4JDgdf5vE0K3YrW9ZnbP'
      }
    ]
  }
];
describe('<Card/>', () => {
  it('renders <Card/> with populated data ', () => {
    const { container, queryByText, getByAltText } = render(
      <Card.Group>
        {slideRow.map((slideItem, i) => (
          <Card key={`${category}${i}-${slideItem?.title?.toLowerCase()}`}>
            <Card.Title>{slideItem?.title}</Card.Title>
            <Card.Entities>
              {slideItem?.data?.map((item) => (
                <Card.Item item={item} key={item.docId}>
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                    alt={item.slug}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src={'/videos/bunny.mp4'} />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );
    expect(queryByText(slideRow[0].title)).toBeTruthy();
    expect(queryByText(slideRow[0].data[0].title)).toBeTruthy();
    expect(getByAltText(slideRow[0].data[0].slug)).toBeTruthy();
    expect(queryByText(slideRow[0].data[0].description)).toBeTruthy();

    expect(queryByText(slideRow[1].title)).toBeTruthy();
    expect(queryByText(slideRow[1].data[0].title)).toBeTruthy();
    expect(getByAltText(slideRow[1].data[0].slug)).toBeTruthy();
    expect(queryByText(slideRow[1].data[0].description)).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('toggles the feature upon clicking  a card item', () => {
    const { queryByText, getByAltText, container, getByTestId } = render(
      <Card.Group>
        {slideRow.map((slideItem, i) => (
          <Card key={`${category}${i}-${slideItem?.title?.toLowerCase()}`}>
            <Card.Title>{slideItem?.title}</Card.Title>
            <Card.Entities>
              {slideItem?.data?.map((item) => (
                <Card.Item item={item} key={item.docId} data-testid={`${item.slug}-item-feature`}>
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                    alt={item.slug}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src={'/videos/bunny.mp4'} />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(queryByText('18')).toBeFalsy();
    fireEvent.click(getByTestId('tiger-king-item-feature'));
    expect(queryByText('18')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('18')).toBeFalsy();

    expect(queryByText('PG')).toBeFalsy();
    fireEvent.click(getByTestId('juno-item-feature'));
    expect(queryByText('PG')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('PG')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
