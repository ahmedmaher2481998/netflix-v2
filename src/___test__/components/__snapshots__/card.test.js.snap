// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders <Card/> with populated data  1`] = `
<div
  class="cards__Group-sc-1y7r0tx-11 jIAYgV"
>
  <div
    class="cards__Container-sc-1y7r0tx-1 JkTZa"
  >
    <p
      class="cards__Title-sc-1y7r0tx-0 fZSMtY"
    >
      Documentaries
    </p>
    <div
      class="cards__Entities-sc-1y7r0tx-12 gTKCYJ"
    >
      <div
        class="cards__Item-sc-1y7r0tx-5 jYycTW"
      >
        <img
          alt="tiger-king"
          class="cards__Image-sc-1y7r0tx-6 bTjcxW"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="cards__Meta-sc-1y7r0tx-3 hFedYa"
        >
          <p
            class="cards__SubTitle-sc-1y7r0tx-2 bwusPp"
          >
            Tiger King
          </p>
          <p
            class="cards__Text-sc-1y7r0tx-4 exhwMV"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="cards__Container-sc-1y7r0tx-1 JkTZa"
  >
    <p
      class="cards__Title-sc-1y7r0tx-0 fZSMtY"
    >
      Feel Good
    </p>
    <div
      class="cards__Entities-sc-1y7r0tx-12 gTKCYJ"
    >
      <div
        class="cards__Item-sc-1y7r0tx-5 jYycTW"
      >
        <img
          alt="juno"
          class="cards__Image-sc-1y7r0tx-6 bTjcxW"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="cards__Meta-sc-1y7r0tx-3 hFedYa"
        >
          <p
            class="cards__SubTitle-sc-1y7r0tx-2 bwusPp"
          >
            Juno
          </p>
          <p
            class="cards__Text-sc-1y7r0tx-4 exhwMV"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card/> toggles the feature upon clicking  a card item 1`] = `
<div
  class="cards__Group-sc-1y7r0tx-11 jIAYgV"
>
  <div
    class="cards__Container-sc-1y7r0tx-1 JkTZa"
  >
    <p
      class="cards__Title-sc-1y7r0tx-0 fZSMtY"
    >
      Documentaries
    </p>
    <div
      class="cards__Entities-sc-1y7r0tx-12 gTKCYJ"
    >
      <div
        class="cards__Item-sc-1y7r0tx-5 jYycTW"
        data-testid="tiger-king-item-feature"
      >
        <img
          alt="tiger-king"
          class="cards__Image-sc-1y7r0tx-6 bTjcxW"
          src="/images/series/documentaries/tiger-king/small.jpg"
        />
        <div
          class="cards__Meta-sc-1y7r0tx-3 hFedYa"
        >
          <p
            class="cards__SubTitle-sc-1y7r0tx-2 bwusPp"
          >
            Tiger King
          </p>
          <p
            class="cards__Text-sc-1y7r0tx-4 exhwMV"
          >
            Tiger King description
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="cards__Container-sc-1y7r0tx-1 JkTZa"
  >
    <p
      class="cards__Title-sc-1y7r0tx-0 fZSMtY"
    >
      Feel Good
    </p>
    <div
      class="cards__Entities-sc-1y7r0tx-12 gTKCYJ"
    >
      <div
        class="cards__Item-sc-1y7r0tx-5 jYycTW"
        data-testid="juno-item-feature"
      >
        <img
          alt="juno"
          class="cards__Image-sc-1y7r0tx-6 bTjcxW"
          src="/images/series/feel-good/juno/small.jpg"
        />
        <div
          class="cards__Meta-sc-1y7r0tx-3 hFedYa"
        >
          <p
            class="cards__SubTitle-sc-1y7r0tx-2 bwusPp"
          >
            Juno
          </p>
          <p
            class="cards__Text-sc-1y7r0tx-4 exhwMV"
          >
            Juno description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
